packages/backend/src/utils/npm.ts

```diff
@@ -0,0 +1,127 @@
+import { exec, execSync } from "child_process";
+import { promisify } from "util";
+import { platform } from "os";
+import * as _ from "lodash";
+import * as customLocation from "./customLocation";
+import * as sudo from "sudo-prompt";
+import { promises, constants } from "fs";
+import messages from "../messages";
+import { vscode } from "./vscodeProxy";
+import * as path from "path";
+
+export const isWin32 = platform() === "win32";
+const NPM = isWin32 ? "npm.cmd" : "npm";
+
+const NPM_REGISTRY_HOST = _.get(process, "env.NPM_CFG_REGISTRY", "http://registry.npmjs.com/");
+const SEARCH_QUERY_PREFIX = `${NPM_REGISTRY_HOST}-/v1/search?text=`;
+const SEARCH_QUERY_SUFFIX = "keywords:yeoman-generator &size=25&ranking=popularity";
+
+const CANCELED = "Action cancelled";
+const HAS_ACCESS = "Has Access";
+
+const enum choices {
+  set_defaul_location,
+  change_global_owner,
+  cancelled,
+}
+
+class Command {
+  private readonly globalNodeModulesPath;
+  private readonly globalPath;
+  private readonly CHANGE_OWNER_FOR_GLOBAL;
+  private readonly SET_DEFAULT_LOCATION;
+
+  constructor() {
+    this.globalNodeModulesPath = _.trim(execSync(`${NPM} root -g`).toString());
+
+    const nmLength = path.join(path.sep, "node_modules").length;
+    this.globalPath = this.globalNodeModulesPath.substring(0, this.globalNodeModulesPath.length - nmLength);
```

Comment:
```suggestion
    this.globalPath = this.globalNodeModulesPath;
```
I had to change as follows to get the the  isPathWritable check to failing locally on mac.

Reply:
globalPath does not contain /node_modules at the end
I did not understand what is wrong with this.
Could you give an example or explain how to reproduce the issue?

Reply:
Weird seems to be working ok now. 🤔